nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/impl/ScalaScriptEngineConfigurator.java

```diff
@@ -0,0 +1,132 @@
+import java.io.UnsupportedEncodingException;
+
+public class ScalaScriptEngineConfigurator implements ScriptEngineConfigurator, Compilable {
+
+    private static final String PRELOADS =
+            "import org.apache.nifi.components._\n"
+                    + "import org.apache.nifi.flowfile.FlowFile\n"
+                    + "import org.apache.nifi.processor._\n"
+                    + "import org.apache.nifi.processor.exception._\n"
+                    + "import org.apache.nifi.processor.io._\n"
+                    + "import org.apache.nifi.processor.util._\n"
+                    + "import org.apache.nifi.processors.script._\n"
+                    + "import org.apache.nifi.processors.script.ExecuteScript._\n"
+                    + "import org.apache.nifi.logging.ProcessorLog\n";
+
+    private ScriptEngine scriptEngine;
+
+    @Override
+    public String getScriptEngineName() {
+        return "Scala";
+    }
+
+    @Override
+    public Object init(ScriptEngine engine, String modulePath) throws ScriptException {
+
+        if (modulePath != null) {
+            final Settings settings = ((IMain) engine).settings();
+
+            // We need to find our NAR's working lib/ directory, that's where the Scala JARs are
+            String myPath = this.getClass().getProtectionDomain().getCodeSource().getLocation().getPath();
+
+            String parent = new File(myPath).getParent();
+            String scalaLib = StringUtils.replace(parent, "\\", "/")+"/*";
+
+            // The usual way to set up the scala compiler is to set java.class.path and then do something like this:
+            //  final MutableSettings.BooleanSetting useJavaClasspath = (MutableSettings.BooleanSetting) settings.usejavacp();
+            //  useJavaClasspath.tryToSetFromPropertyValue("true");
+            //
+            // However it appears if you set it up to use the java classpath, it won't find the rest of the stuff we
+            // need (namely the NAR's dependencies.  So instead we leave the above setting "false" and build the
+            // classpath ourselves, prepending the NAR's dependency folder to the value of java.class.path
+            final MutableSettings.PathSetting classpath = settings.classpath();
+            classpath.prepend(System.getProperty("java.class.path"));
+            classpath.prepend(scalaLib);
+
+
+        }
+        scriptEngine = engine;
+        return scriptEngine;
+    }
+
+    @Override
+    public Object eval(ScriptEngine engine, InputStream scriptStream, String modulePath) throws ScriptException {
+        try {
+            scriptEngine = engine;
+            ByteArrayInputStream bais = new ByteArrayInputStream(PRELOADS.getBytes("UTF-8"));
+            SequenceInputStream fullInputStream = new SequenceInputStream(bais, scriptStream);
+            return engine.eval(new InputStreamReader(fullInputStream));
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
```

Comment:
I think this one slipped through the cracks - should this throw ScriptException?